koku/masu/processor/tasks.py

```diff
@@ -85,59 +86,64 @@ def get_report_files(
         self, customer_name, authentication, billing_source, provider_type, provider_uuid, month, cache_key
     )
 
-    try:
-        stmt = (
-            f"Reports to be processed:\n"
-            f" schema_name: {customer_name}\n"
-            f" provider: {provider_type}\n"
-            f" provider_uuid: {provider_uuid}\n"
-        )
-        for report in reports:
-            stmt += " file: " + str(report["file"]) + "\n"
-        LOG.info(stmt[:-1])
-        reports_to_summarize = []
-        for report_dict in reports:
-            manifest_id = report_dict.get("manifest_id")
-            file_name = os.path.basename(report_dict.get("file"))
-            with ReportStatsDBAccessor(file_name, manifest_id) as stats:
-                started_date = stats.get_last_started_datetime()
-                completed_date = stats.get_last_completed_datetime()
-
-            # Skip processing if already in progress.
-            if started_date and not completed_date:
-                expired_start_date = started_date + datetime.timedelta(hours=Config.REPORT_PROCESSING_TIMEOUT_HOURS)
-                if DateAccessor().today_with_timezone("UTC") < expired_start_date:
-                    LOG.info(
-                        "Skipping processing task for %s since it was started at: %s.", file_name, str(started_date)
-                    )
-                    continue
-
+    with transaction.atomic():
```

Comment:
I think I'm regretting agreeing to this idea. If I'm understanding this idea correctly, we are processing the _entire_ manifest within an atomic block. Let's say we had a particularly large manifest, say, 50 files, each ridiculously huge. And then, let's suppose we lost our DB connection whilst processing the 50th file, raising a `cursor already closed` error. Wouldn't this atomic block mean that we would have to process every file a second time?

Reply:
yea thats a good point.  Maybe we should move the atomic block to be after  `for report_dict in reports:`.  That way we would only have to re-process the report (file) that failed.